ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml

```diff
@@ -2,57 +2,111 @@ import QtQuick 2.14
+            width: 40
+            height: 40
+            name: notification.message.senderIcon || ""
+            assetSettings.isImage: notification.message.senderIcon.startsWith("data")
+            pubkey: notification.message.senderId
+            colorId: Utils.colorIdForPubkey(notification.message.senderId)
+            colorHash: Utils.getColorHashAsJson(notification.message.senderId, false, true)
+            showRing: true
+        }
+    }
+
+    property Component messageSubheaderComponent: null
+    property Component messageBadgeComponent: null
+
+    function openProfilePopup() {
+        activityCenterClose()
+        Global.openProfilePopup(notification.message.senderId)
+    }
+
+    bodyComponent: RowLayout {
+        id: messageRow
+        spacing: 8
+
+        Item {
+            implicitWidth: root.messageDetails.sender.profileImage.assetSettings.width
+            implicitHeight: profileImage.height
+            Layout.alignment: Qt.AlignTop
+            Layout.topMargin: 2
+
+            StatusSmartIdenticon {
+                id: profileImage
+                name: root.messageDetails.sender.displayName
+                asset: root.messageDetails.sender.profileImage.assetSettings
+                ringSettings: root.messageDetails.sender.profileImage.ringSettings
+
+                MouseArea {
+                    cursorShape: enabled ? Qt.PointingHandCursor : Qt.ArrowCursor
+                    acceptedButtons: Qt.LeftButton | Qt.RightButton
+                    anchors.fill: parent
+                    onClicked: root.openProfilePopup()
+                }
             }
+        }
+
+        ColumnLayout {
+            spacing: 2
+            Layout.alignment: Qt.AlignTop
+            Layout.fillWidth: true
 
-            root.activityCenterStore.switchTo(notification)
-            root.activityCenterClose()
+            StatusMessageHeader {
+                sender: root.messageDetails.sender
+                amISender: root.messageDetails.amISender
+                messageOriginInfo: root.messageDetails.messageOriginInfo
+                timestamp.text: root.timestampString
+                timestamp.tooltip.text: root.timestampTooltipString
+                onClicked: root.openProfilePopup()
+            }
+
+            Loader {
+                sourceComponent: root.messageSubheaderComponent
+                Layout.fillWidth: true
+            }
+
+            RowLayout {
+                spacing: 2
+                Layout.fillWidth: true
+
+                StatusBaseText {
+                    text: Utils.stripHtmlTags(root.messageDetails.messageText)
+                    maximumLineCount: root.maximumLineCount
+                    wrapMode: Text.Wrap
+                    elide: Text.ElideRight
+                    font.pixelSize: 15
+                    Layout.alignment: Qt.AlignVCenter
+                    Layout.maximumWidth: 400 // From designs, fixed value to align all possible CTAs
```

Comment:
You need to set `Layout.preferredWidth` for the `wrap/elide` to work

Reply:
That works fine with Layout.maximumWidth
<img width="543" alt="Screenshot 2022-11-17 at 18 12 40" src="https://user-images.githubusercontent.com/2522130/202468982-70aafc46-2e58-4d13-b0d1-16f77ba8d205.png">

Reply:
Cool!